pkg/build/controller/build/build_controller.go

```diff
@@ -2023,25 +2067,41 @@ func (bc *BuildController) getAdditionalTrustedCAData(config *configv1.Image) (m
 	return additionalCA.Data, nil
 }
 
-func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image) (string, error) {
+func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image, policy *operatorv1alpha1.ImageContentSourcePolicy) (string, error) {
 	registriesConfig := config.Spec.RegistrySources
-	if len(registriesConfig.InsecureRegistries) == 0 {
+	if len(registriesConfig.InsecureRegistries) == 0 && (policy == nil || len(policy.Spec.RepositoryDigestMirrors) == 0) {
 		klog.V(4).Info("using default insecure registry settings for builds")
 		return "", nil
 	}
+	mirrors := []operatorv1alpha1.RepositoryDigestMirrors{}
+	if policy != nil {
+		mirrors = policy.Spec.RepositoryDigestMirrors
+	}
+	regs := []registry{}
+	defaultReg := registry{}
+	// docker.io must be the only entry in the registry search list
+	// See https://github.com/openshift/builder/pull/40
+	defaultReg.Location = "docker.io"
+	regs = append(regs, defaultReg)
+	for _, regStr := range registriesConfig.InsecureRegistries {
+		reg := registry{}
+		reg.Location = regStr
+		reg.Insecure = true
+		regs = append(regs, reg)
+	}
+	for _, mirror := range mirrors {
+		reg := registry{}
+		reg.Prefix = mirror.Source
+		reg.Mirrors = []endpoint{}
```

Comment:
`reg` is not used/appended anywhere?

In addition, the configuration in `registries.conf` must not contain an entry for the same `Location` twice, so merely appending does not work.